src/nni_evaluation_engine.hpp

```diff
@@ -0,0 +1,138 @@
+// Copyright 2019-2021 bito project contributors.
+// bito is free software under the GPLv3; see LICENSE file for details.
+//
+// NNI Evaluation Engine
+// 
+
+#pragma once
+
+#include "gp_engine.hpp"
+#include "gp_dag.hpp"
+
+#include "bitset.hpp"
+#include "subsplit_dag.hpp"
+#include "subsplit_dag_nni.hpp"
+#include "graft_dag.hpp"
+#include "sugar.hpp"
+
+class NNIEvaluationEngine {
+ public:
+  // Constructors:
+  NNIEvaluationEngine(GPDAG &dag, GPEngine &gp_engine);
+
+  // ** Getter/Setters
+
+  // Get Reference of DAG.
+  const GPDAG &GetGPDAG() const { return dag_; };
+  // Get Reference of GraftDAG.
+  GraftDAG &GetGraftDAG() const { return *graft_dag_.get(); };
+  // Get Reference of GPEngine.
+  const GPEngine &GetGPEngine() const { return gp_engine_; }
+  // Get Adjacent NNIs to DAG.
+  const SetOfNNIs &GetAdjacentNNIs() { return adjacent_nnis_; };
+  // Get Number of Adjacent NNIs.
+  size_t GetAdjacentNNICount() const { return adjacent_nnis_.GetSize(); };
+  // Get ith NNI from Adjacent NNIs.
+  NNIOperation GetNNIFromAdjacentNNIs(size_t i) const { return adjacent_nnis_.At(i); };
+  // Get Accepted NNIs to DAG.
+  const std::vector<NNIOperation> &GetAcceptedNNIs() { return accepted_nnis_; };
+  // Get Number of Accepted NNIs.
+  size_t GetAcceptedNNICount() const { return accepted_nnis_.size(); };
+  // Get ith NNI from Accepted NNIs.
+  NNIOperation GetNNIFromAcceptedNNIs(size_t i) const { return accepted_nnis_.at(i); };
+  // Get map of ranked NNIs with NNI to score.
+  const std::map<NNIOperation, double> &GetRankedNNIs() { return ranked_nnis_; };
+  // Get number of runs of NNI engine.
+  size_t GetNumberOfPasses() { return number_passes_; };
+
+  // ** Runners
+  // These start the engine, which procedurally ranks and adds (and
```

Comment:
After reading the implementation, I see what's going on here, but I think it'd be nice to move these docs to the top of the header file and expand a little bit. They really define what the whole class is about.